src/db/db/gsiDeclDbLibrary.cc

```diff
@@ -874,6 +919,19 @@ Class<db::PCellParameterDeclaration> decl_PCellParameterDeclaration ("db", "PCel
     "This method will add the given value with the given description to the list of\n"
     "choices. If choices are defined, KLayout will show a drop-down box instead of an\n"
     "entry field in the parameter user interface.\n"
+    "If a range is already set for this parameter the choice will not be added and a warning message is showed.\n"
+  ) +
+  gsi::method_ext ("set_range", &set_range, gsi::arg ("low"), gsi::arg ("high"), gsi::arg ("resolution"), gsi::arg ("action"),
+    "@brief Set a range constraint\n"
+    "This method will set a range constraint to the parameter with 'low' and 'high' as minimum and maximum value.\n"
+    "This range constraint will only be set if the parameter-, the low- and the high-type are numeric.\n"
+    "If a choice is already set for this parameter the range will not be set and a warning message is showed.\n"
+    "The optional parameter 'resolution' will give a desired resolution value (currently not used).\n"
+    "The optional parameter 'action' determines the action to be invoked.\n"
+    "This action can be one of three values: REJECT, ACCEPT, USE_DEFAULT. If this failure action\n"
+    "parameter is not specified, then it will be REJECT by default.\n"
```

Comment:
There is no default given.

Reply:
Coming from PyCell API. Will be removed.